src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5379,7 +5384,13 @@ String replaceParameterMarkers(String sqlSrc,
 
         int paramIndex = 0;
         while (true) {
-            int srcEnd = ParameterUtils.scanSQLForChar('?', sqlSrc, srcBegin);
+            int srcEnd;
+            if (paramIndex >= paramPositions.length) {
+                srcEnd = sqlSrc.length();
+            }
+            else {
+                srcEnd = paramPositions[paramIndex];
+            }
```

Comment:
5388 - 5393 can be replaced with
`srcEnd = (paramIndex >= paramPositions.length) ? sqlSrc.length() : paramPositions[paramIndex];`